pkg/resolver/searchSchema.go

```diff
@@ -0,0 +1,75 @@
+package resolver
+
+import (
+	"context"
+
+	"github.com/doug-martin/goqu/v9"
+	"github.com/driftprogramming/pgxpoolmock"
+	db "github.com/stolostron/search-v2-api/pkg/database"
+	klog "k8s.io/klog/v2"
+)
+
+type SearchSchemaMessage struct {
+	pool   pgxpoolmock.PgxPool
+	query  string
+	params []interface{}
+}
+
+func SearchSchema(ctx context.Context) (map[string]interface{}, error) {
+	searchSchemaResult := &SearchSchemaMessage{
+		pool: db.GetConnection(),
+	}
+	searchSchemaResult.searchSchemaQuery(ctx)
+	return searchSchemaResult.searchSchemaResults()
+}
+
+func (s *SearchSchemaMessage) searchSchemaQuery(ctx context.Context) {
+	var selectDs *goqu.SelectDataset
+
+	// schema query sample: "SELECT distinct jsonb_object_keys(jsonb_strip_nulls(data)) FROM search.resources"
```

Comment:
This query doesn't show keys with null values but keys with empty string values are not excluded.